src/test/scala/io/qbeast/spark/delta/hook/PreCommitHookIntegrationTest.scala

```diff
@@ -0,0 +1,88 @@
+package io.qbeast.spark.delta.hook
+
+import io.qbeast.context.QbeastContext
+import io.qbeast.core.model.QTableID
+import io.qbeast.spark.delta.hook.PreCommitHook.PRE_COMMIT_HOOKS_PREFIX
+import io.qbeast.spark.delta.hook.PreCommitHook.PreCommitHookOutput
+import io.qbeast.spark.delta.hook.SimpleHook.testOutput
+import io.qbeast.spark.QbeastIntegrationTestSpec
+import org.apache.spark.sql.delta.actions.Action
+import org.apache.spark.sql.delta.actions.CommitInfo
+import org.apache.spark.sql.delta.util.FileNames
+import org.apache.spark.sql.delta.DeltaLog
+
+private class SimpleHook extends PreCommitHook {
+  override val name: String = "SimpleHook"
+
+  override def run(args: Seq[Action]): PreCommitHookOutput = {
+    testOutput
+  }
+
+}
+
+private object SimpleHook {
+  val testOutput: Map[String, String] = Map("clsName" -> "SimpleHook")
+}
+
+class PreCommitHookIntegrationTest extends QbeastIntegrationTestSpec {
```

Comment:
Should we test as well the execution of several precommit hooks?

Reply:
Done!